src/osd/PG.cc

```diff
@@ -1870,7 +1870,9 @@ bool PG::op_has_sufficient_caps(OpRequestRef& op)
 			     op->need_read_cap(),
 			     op->need_write_cap(),
 			     op->need_class_read_cap(),
-			     op->need_class_write_cap());
+			     op->need_class_write_cap(),
+			     op->class_name(),
+			     op->class_whitelisted());
```

Comment:
The librados ObjectOperation api lets you perform multiple execs in the same Operation.